apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx

```diff
@@ -62,6 +60,13 @@ const obtainNewUsernameChangeCondition = ({
   if (!userIsPremium && isNewUsernamePremium && !stripeCustomer?.paidForPremium) {
     return UsernameChangeStatusEnum.UPGRADE;
   }
+
+  // This code requires payment from already premium usernames(if they don't have payment info in stripe under paidForPremium)
+  // If we are sure that metadata.isPremium is set to true for all existing premium users we can uncomment this code with an additional check of `&& !metadata.isPremium`
+  // if (userIsPremium && isNewUsernamePremium && !stripeCustomer?.paidForPremium) {
+  //   return UsernameChangeStatusEnum.UPGRADE;
+  // }
```

Comment:
I think there were several attempts to achieve this, so to my knowledge everyone should have it already.
What we did back then was:
- Get list of premium users on STRIPE
- Add metadata key to the user DB with isPremium: true

Reply:
Also every time a subscription was renewed or the username was trying to downgrade we always read from stripe for highest subscription tier active for the single user.

Reply:
Thanks @alannnc. I have started rechecking user.metadata.isPremium to decide if payment is required or not.
I am not aware of the entire flow of Stripe payments and webhooks. Can you please test it at your end now.